hooks/environment

```diff
@@ -12,6 +12,25 @@ check_command() {
   fi
 }
 
+# In `buildkite-agent` 3.66.0
+# (https://github.com/buildkite/agent/releases/tag/v3.66.0) the ability to
+# redact secrets (https://buildkite.com/docs/agent/v3/cli-redactor) was added.
+# However, because we may not always run that version of `buildkite-agent`, we
+# should conditionally enable it
+check_buildkite_version() {
+  local version_output
+  version_output=$(buildkite-agent --version)
+  local version
+  version=$(echo "$version_output" | grep -oP '\d+\.\d+\.\d+')
+  local min_version="3.66.0"
+  if [ "$(printf '%s\n' "$min_version" "$version" | sort -V | head -n1)" = "$min_version" ]; then
```

Comment:
I've definitely been confused by the `sort -V` before, so it may be worth using the long flag name:

```suggestion
  if [ "$(printf '%s\n' "$min_version" "$version" | sort --version-sort | head -n1)" = "$min_version" ]; then
```

Reply:
changed in 2cab9997094d5f493b5fd50e2ac765f66a5e37de